### DISCUSSION TITLE
Error: remote room (S-RWCbe1Y) timed out

### DISCUSSION THREAD

**Discussion by kelvinlee** [colyseus/colyseus#600] [Q&A] [answered]
After I open a room, I receive a large amount of data from the outside. 
When there is a significant amount of data, I encounter an error:

```
ServerError: remote room (S-RWCbe1Y) timed out, requesting "onNewMessage with args [{"type":"live_comment","content":{"msg_id":"123","sec_openid":"123","content":"2","avatar_url":"123","nickname":"123","timestamp":1688877442000}}]". (2000ms exceeded)
    at Object.remoteRoomCall (/my/fire-to-building-backend/node_modules/@colyseus/core/build/MatchMaker.js:177:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 4213
}
```

The data is quite large, and approximately 500 pieces of data are being sent every second. Has anyone encountered this issue before and knows how to resolve it?

The code how was I send the message.
```
                 var rooms = await matchMaker.query({"name": roomid});
		if (rooms.length>0) {
			var room = rooms[0]
			if (room) matchMaker.remoteRoomCall(room.roomId, "onNewMessage", [message])
		}else{
			console.log("dont have the room : " + roomid)
		}
```

**Comment by endel** [comment, marked as answer]
Hi @kelvinlee, each remote room call will generate two messages under the hood (one for request, other for the response) and should be used with caution. 

By your code it seems you could achieve the same result by using publish/subscribe instead, without the need to query all the rooms, and reducing the amount of messages going through Redis. 

Hope this helps, if it doesn’t, please share your use-case as it may be solved differently in a more efficient way. Cheers

**Reply by kelvinlee** [reply]
Does publish/subscribe mean "presence"?
When I publish a topic, will all rooms receive the same message?

**Reply by kelvinlee** [reply]
I understand it this way, right?
I use subscription in room.
```
this. presence. subscribe("onNewMessage", (data) => {
    console.log("received message:", data);
});
```
and then use externally
```
matchMaker.presence.publish("onNewMessage", {"type":"live_comment","content":{"msg_id":"123","sec_openid":"123","content":"2","avatar_url ":"123","nickname":"123","timestamp":1688877442000}});
```
So I have a question, if I have multiple rooms
Can I specify the room message by modifying the subscribed topic, for example:
```
// 123 is the roomName
this.presence.subscribe(this.roomName+"-onNewMessage", (data) => {
    console.log("received message:", data);
});
```
then
```
matchMaker.presence.publish("123-onNewMessage", {"type":"live_comment","content":{"msg_id":"123","sec_openid":"123","content":"2", "avatar_url":"123","nickname":"123","timestamp":1688877442000}});
```

@endel I am right? or other suggestion?
